sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataReaderDAO.java

```diff
@@ -270,16 +235,19 @@ public void onFailure(Throwable t) {
 
         return new CachingRowGroupIterator(deltaRowGroupResultSetIterator, _driverConfig.getRecordCacheSize(), _driverConfig.getRecordSoftCacheSize());
     }
-
     /**
      * Creates a Record instance for a given key and list of rows.  All rows must be from the same Cassandra row;
      * in other words, it is expected that row.getBytesUnsafe(ROW_KEY_RESULT_SET_COLUMN) returns the same value for
      * each row in rows.
      */
     private Record newRecordFromCql(Key key, Iterable<Row> rows) {
-        Iterator<Map.Entry<UUID, Change>> changeIter = decodeChangesFromCql(rows.iterator());
-        Iterator<Map.Entry<UUID, Compaction>> compactionIter = decodeCompactionsFromCql(rows.iterator());
-        Iterator<RecordEntryRawMetadata> rawMetadataIter = rawMetadataFromCql(rows.iterator());
+        Session session = ((AstyanaxTable) key.getTable()).getReadStorage().getPlacement().getKeyspace().getCqlSession();
```

Comment:
Although inferring the placement from the read storage is likely always correct it's safer to use the placement provided in the upstream call.  I'd recommend changing `newRecordFromCql` to take a `DeltaPlacement` parameter, then using `deltaPlacement.getKeyspace().getCqlSession()`.  It looks like all callers to this method either directly get one or have one from their caller.